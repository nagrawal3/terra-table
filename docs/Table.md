# Terra Table

 The terra table component provides a responsive way to present tabular data.

## Getting Started

- Install with [npmjs](https://www.npmjs.com): 
  - `npm install terra-table` 
  - `yarn install terra-table` 
- [Download the latest version](https://github.com/cerner/terra-table/archive/master.zip)
- Clone the repo: `git clone https://github.com/cerner/terra-table.git`
