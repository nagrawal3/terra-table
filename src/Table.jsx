import React, { PropTypes } from 'react';
import classNames from 'classnames';
import TableHeader from '../src/TableHeader';
import TableHeaderContent from '../src/TableHeaderContent';
import TableRows from '../src/TableRows';
import TableRow from '../src/TableRow';
import TableRowContent from '../src/TableRowContent';
import TableSingleSelectableRows from '../src/SingleSelectableRows';
import '../src/table.scss';

const propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
  isStripped: PropTypes.bool
};

const defaultProps = {
  isStripped: true
};

const Table = ({
  children,
  isStripped,
  ...customProps
  }) => {
  let tableClassNames = classNames([
      'terra-Table',
      {'terra-Table--striped': isStripped},
      customProps.className
    ]);
  return (
    <table {...customProps} className={tableClassNames}>
      {children}
    </table>
  );
};

Table.propTypes = propTypes;
Table.defaultProps = defaultProps;
Table.Rows = TableRows;
Table.Header = TableHeader;
Table.HeaderContent = TableHeaderContent;
Table.Row = TableRow;
Table.RowContent = TableRowContent;
Table.SingleSelectableRows = TableSingleSelectableRows;

export default Table;
