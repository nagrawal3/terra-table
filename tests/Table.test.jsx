import React from 'react';
import initStoryshots from 'storyshots';
import Table from '../src/Table';

const defaultVariety =
(<Table>
  <Table.Header>
    <Table.HeaderContent display={'Column Heading 1 Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={1} minWidth={'huge'} />
    <Table.HeaderContent display={'Column Heading 2'} key={2} minWidth={'medium'} />
    <Table.HeaderContent display={'Column Heading 3 Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={3} minWidth={'large'} />
  </Table.Header>
  <Table.SingleSelectableRows>
    <Table.Row isSelected isSelectable={false} key={0}>
      <Table.RowContent display={'Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={1} />
      <Table.RowContent display={'Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={2} />
      <Table.RowContent display={'Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={3} />
    </Table.Row>
    <Table.Row isSelected key={1}>
      <Table.RowContent display={'Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={1} />
      <Table.RowContent display={'Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={2} />
      <Table.RowContent display={'Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={3} />
    </Table.Row>
    <Table.Row isSelected={false} key={2}>
      <Table.RowContent display={'Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={1} />
      <Table.RowContent display={'Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={2} />
      <Table.RowContent display={'Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text'} key={3} />
    </Table.Row>
  </Table.SingleSelectableRows>
</Table>);

// Run snapshot tests for react-storybook
initStoryshots();

// Snapshot Tests
it('should render a default component', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper).toMatchSnapshot();
});

// Prop Tests
it('should have the class terra-Table--default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper.prop('className')).toContain('terra-Table');
});
