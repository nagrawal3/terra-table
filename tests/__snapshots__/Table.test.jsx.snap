exports[`Storyshots Table  An example of the base table with long string 1`] = `
<span>
  <table
    className="terra-Table terra-Table--striped">
    <thead>
      <tr>
        <th
          className="terra-Table-width-huge"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 1 astringofreallylongtexttotestalongsinglewordtestcase
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-medium"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 2 astringofreallylongtexttotestalongsinglewordtestcase
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-large"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 3 astringofreallylongtexttotestalongsinglewordtestcase
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="terra-Table--isSelected">
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
      </tr>
      <tr
        className="">
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
      </tr>
      <tr
        className="">
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</span>
`;

exports[`Storyshots Table  An example of the base table with long text. Rows will have vertical scroll bar 1`] = `
<span>
  <table
    className="terra-Table terra-Table--striped">
    <thead>
      <tr>
        <th
          className="terra-Table-width-huge"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 1 Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-medium"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 2
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-large"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 3 Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="terra-Table--isSelected">
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
      </tr>
      <tr
        className="">
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
      </tr>
      <tr
        className="">
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={undefined}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</span>
`;

exports[`Storyshots Table  An example of the non-selectable rows in singleselectable table 1`] = `
<span>
  <table
    className="terra-Table terra-Table--striped">
    <thead>
      <tr>
        <th
          className="terra-Table-width-huge"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column 1
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-medium"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 2
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-large"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 3
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 1
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 2
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 3
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelected terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 1
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 2
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 3
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 1
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 2
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 3
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 1
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 2
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 3
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 1
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 2
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 3
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</span>
`;

exports[`Storyshots Table  An example of the selectable base table with long string 1`] = `
<span>
  <table
    className="terra-Table terra-Table--striped">
    <thead>
      <tr>
        <th
          className="terra-Table-width-huge"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 1 astringofreallylongtexttotestalongsinglewordtestcase
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-medium"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 2 astringofreallylongtexttotestalongsinglewordtestcase
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-large"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 3 astringofreallylongtexttotestalongsinglewordtestcase
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="terra-Table--isSelected terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-small terra-Table-content">
            astringofreallylongtexttotestalongsinglewordtestcase
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</span>
`;

exports[`Storyshots Table  An example of the selectable base table with long text 1`] = `
<span>
  <table
    className="terra-Table terra-Table--striped">
    <thead>
      <tr>
        <th
          className="terra-Table-width-huge"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 1 Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-medium"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 2
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-large"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 3 Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelected terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</span>
`;

exports[`Storyshots Table  An example of the selectable base table with long text. Row and header height is fixed. Add vertical scroll 1`] = `
<span>
  <table
    className="terra-Table terra-Table--striped">
    <thead
      isSelectable={true}>
      <tr>
        <th
          className="terra-Table-width-huge"
          onClick={undefined}>
          <div
            className="terra-Table-headerHeight-medium terra-Table-content">
            Column Heading 1 Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-medium"
          onClick={undefined}>
          <div
            className="terra-Table-headerHeight-medium terra-Table-content">
            Column Heading 2
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-large"
          onClick={undefined}>
          <div
            className="terra-Table-headerHeight-medium terra-Table-content">
            Column Heading 3 Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="terra-Table--isSelected terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-large terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-large terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-large terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-large terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-large terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-large terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-large terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-large terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-large terra-Table-content">
            Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</span>
`;

exports[`Storyshots Table  An example of the selectable base table without zebra stipping 1`] = `
<span>
  <table
    className="terra-Table">
    <thead>
      <tr>
        <th
          className="terra-Table-width-huge"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column 1
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-medium"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 2
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
        <th
          className="terra-Table-width-large"
          onClick={undefined}>
          <div
            className="terra-Table-content">
            Column Heading 3
            <span
              className="terra-Table-hidden">
              ↑
            </span>
            <span
              className="terra-Table-hidden">
              ↓
            </span>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="terra-Table--isSelected terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 1
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 2
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 3
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 1
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 2
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 3
          </div>
        </td>
      </tr>
      <tr
        className="terra-Table--isSelectable">
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 1
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 2
          </div>
        </td>
        <td
          onClick={[Function]}>
          <div
            className="terra-Table-height-medium terra-Table-content">
            Column Text 3
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</span>
`;

exports[`test should render a default component 1`] = `
<table
  className="terra-Table terra-Table--striped">
  <TableHeader>
    <TableHeaderContent
      display="Column Heading 1 Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text"
      minWidth="huge" />
    <TableHeaderContent
      display="Column Heading 2"
      minWidth="medium" />
    <TableHeaderContent
      display="Column Heading 3 Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text"
      minWidth="large" />
  </TableHeader>
  <SingleSelectableRows>
    <TableRow
      className=""
      isSelectable={false}
      isSelected={true}>
      <TableRowContent
        display="Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text" />
      <TableRowContent
        display="Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text" />
      <TableRowContent
        display="Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text" />
    </TableRow>
    <TableRow
      className=""
      isSelected={true}>
      <TableRowContent
        display="Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text" />
      <TableRowContent
        display="Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text" />
      <TableRowContent
        display="Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text" />
    </TableRow>
    <TableRow
      className=""
      isSelected={false}>
      <TableRowContent
        display="Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text" />
      <TableRowContent
        display="Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text" />
      <TableRowContent
        display="Vey long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text Very long button text" />
    </TableRow>
  </SingleSelectableRows>
</table>
`;
